src/js/https_client.js

```diff
@@ -0,0 +1,185 @@
+  this.ca = options.ca || '';
+  this.cert = options.cert || '';
+  this.key = options.key || '';
+
+  var isMethodGood = false;
+  for (var key in methods) {
+    if (methods.hasOwnProperty(key)) {
+      if(this.method == methods[key]) {
+        isMethodGood = true;
+      }
+    }
+  }
+
+  if(!isMethodGood) {
+    var err = new Error('Incorrect options.method.')
+    this.emit('error', err);
+    return;
+  }
+
+  this._incoming = new incoming.IncomingMessage(this);
+  this._incoming.url = this.host;
+  this._incoming.method = this.method;
+  this.aborted = null;
+
+  // Register response event handler.
+  if (cb) {
+    this.once('response', cb);
+  }
+  this.once('finish', this.onFinish);
+
+  httpsNative.createRequest(this);
+
+  if (options.auth) {
+    var headerString = 'Authorization: Basic ' + toBase64(options.auth);
+    httpsNative.addHeader(headerString, this);
+  }
+  if (options.headers) {
+    var keys = Object.keys(options.headers);
+    for (var i = 0, l = keys.length; i < l; i++) {
+      var key = keys[i];
+      httpsNative.addHeader(key + ': ' + options.headers[key], this);
+    }
+  }
+  httpsNative.sendRequest(this);
+}
+
+util.inherits(ClientRequest, stream.Writable);
+
+// Concrete stream overriding the empty underlying _write method.
+ClientRequest.prototype._write = function(chunk, callback, onwrite) {
+  httpsNative._write(this, chunk.toString(), callback, onwrite);
+};
+
+ClientRequest.prototype.headersComplete = function() {
+  var self = this;
+  self.emit('response', self._incoming);
+  var isHeadResponse = (self.method == 'HEAD');
+  return isHeadResponse;
+};
+
+ClientRequest.prototype.onError = function(ret) {
+  this.emit('error', ret);
+};
+
+ClientRequest.prototype.onFinish = function() {
+  httpsNative.finishRequest(this);
+};
+
+ClientRequest.prototype.setTimeout = function(ms, cb) {
+  this._incoming.setTimeout(ms, cb);
+};
+
+ClientRequest.prototype.abort = function(doNotEmit) {
+  if (!this.aborted) {
+    httpsNative.abort(this);
+    var date = new Date();
+    this.aborted = date.getTime();
+
+    if (this._incoming.parser) {
+      this._incoming.parser.finish();
+      this._incoming.parser = null;
+    }
+
+    if (!doNotEmit) {
+      this.emit('abort');
+    }
+  }
+};
+
+exports.ClientRequest = ClientRequest;
+
+//Function for encoding options.auth
+function utf8Encode(string) {
+  string = string.replace(/\r\n/g, '\n');
+  var utftext = '';
+
+  for (var n = 0; n < string.length; n++) {
+
+    var c = string.charCodeAt(n);
+
+    if (c < 128) {
+      utftext += String.fromCharCode(c);
+    } else if ((c > 127) && (c < 2048)) {
+      utftext += String.fromCharCode((c >> 6) | 192);
+      utftext += String.fromCharCode((c & 63) | 128);
+    } else {
+      utftext += String.fromCharCode((c >> 12) | 224);
+      utftext += String.fromCharCode(((c >> 6) & 63) | 128);
+      utftext += String.fromCharCode((c & 63) | 128);
+    }
+
+  }
+
+  return utftext;
```

Comment:
Going by a new example in the Buffer API documentation, it would appear I can indeed use the above method. I'll remove this function utf8Encode and use buffers for this.